examples/platforms/simulation/ble.c

```diff
@@ -26,50 +26,204 @@
+                                    uint16_t                aHandle,
+                                    const otBleRadioPacket *aPacket) // wysyłanie
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    OT_UNUSED_VARIABLE(aHandle);
+
+    ssize_t rval;
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(sFd != -1, error = OT_ERROR_INVALID_STATE);
+
+    rval = sendto(sFd, (const char *)aPacket->mValue, 1 + aPacket->mLength, 0, (struct sockaddr *)&sSockaddr,
+                  sizeof(sSockaddr));
+
+    if (rval)
+    {
+        printf("function: %s line: %d\n", __func__, __LINE__);
+    }
+
+exit:
+    return error;
 }
 
-otError otPlatBleGattServerIndicate(otInstance *aInstance, uint16_t aHandle, const otBleRadioPacket *aPacket)
+void platformBleDeinit(void) { deinitFds(); }
+
+void platformBleUpdateFdSet(fd_set *aReadFdSet, fd_set *aWriteFdSet, struct timeval *aTimeout, int *aMaxFd)
+{
+    OT_UNUSED_VARIABLE(aTimeout);
+    OT_UNUSED_VARIABLE(aWriteFdSet);
+
+    if (aReadFdSet != NULL && sFd != -1)
+    {
+        FD_SET(sFd, aReadFdSet);
+
+        if (aMaxFd != NULL && *aMaxFd < sFd)
+        {
+            *aMaxFd = sFd;
+        }
+    }
+}
+
+void platformBleProcess(otInstance *aInstance, const fd_set *aReadFdSet, const fd_set *aWriteFdSet)
+{
+    OT_UNUSED_VARIABLE(aReadFdSet);
+    OT_UNUSED_VARIABLE(aWriteFdSet);
+
+    if (sFd == -1)
+    {
+        return;
+    }
+
+    if (FD_ISSET(sFd, aReadFdSet))
+    {
+        socklen_t len = sizeof(sSockaddr);
+        ssize_t   rval;
+
+        memset(&sSockaddr, 0, sizeof(sSockaddr)); // last that sends wins is it correct??
+        rval = recvfrom(sFd, otPlatBleMsgBuf, sizeof(otPlatBleMsgBuf), 0, (struct sockaddr *)&sSockaddr, &len);
+
+        otPlatBleMsgBuf[len] = 0;
+
+        fprintf(stderr, "%s\n", otPlatBleMsgBuf);
+
+        if (rval > 0)
+        {
+            otBleRadioPacket myPacket;
+            myPacket.mValue  = otPlatBleMsgBuf;
+            myPacket.mLength = (uint16_t)rval;
+            myPacket.mPower  = 0;
+            otPlatBleGattServerOnWriteRequest(
+                aInstance, 0,
+                &myPacket); // TO DO consider passing otPlatBleGattServerOnWriteRequest as a callback function
+        }
+        else if (rval == 0)
+        {
+            // socket is closed, which should not happen
+            assert(false);
+        }
+        else if (errno != EINTR && errno != EAGAIN)
+        {
+            perror("recvfrom(sFd)");
+            exit(EXIT_FAILURE);
+        }
+    }
+}
+
+OT_TOOL_WEAK void otPlatBleGattServerOnWriteRequest(otInstance             *aInstance,
+                                                    uint16_t                aHandle,
+                                                    const otBleRadioPacket *aPacket)
 {
     OT_UNUSED_VARIABLE(aInstance);
     OT_UNUSED_VARIABLE(aHandle);
     OT_UNUSED_VARIABLE(aPacket);
-    return OT_ERROR_NOT_IMPLEMENTED;
+
+    // In case of rcp
```

Comment:
It looks like the comment is trying to explain why this function is added here. Since it's already implemented in `ble_secure.cpp`, I'm guessing there are some circumstances when that other code does not get included in the build.
But from the comment it's not clear what these circumstances are exactly. Maybe the comment should be expanded to explain this? E.g. is it because an RCP build will have `OPENTHREAD_CONFIG_BLE_TCAT_ENABLE == 0` ?

Reply:
@EskoDijk There is a slight problem with libs that I have noticed here. 
the ot-rpc is supposed to be independent of eg FTD/MTD libs. But usage of `otPlatBleGattServerOnWriteRequest` in `platformBleProcess` introduces dependency to Those parts.
Normally while building RCP it's not an issue since TCAT feature is disabled there but in simulation we are enabling it globally which creates mentioned dependency.  Creating weak symbol allows me to workaround this issue till I'll figure out how to rework the cmake linking/dependencies for TCAT in simulation.

Reply:
Ok, sounds complex - in this case better make the comment a TODO explaining this.